internal/app/handlers.go

```diff
@@ -0,0 +1,74 @@
+package app
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/Longreader/go-shortener-url.git/internal/shortener" //indirect
+	"github.com/Longreader/go-shortener-url.git/internal/storage"
+	"github.com/go-chi/chi/v5"
+)
+
+var Store = storage.New()
+
+func IDGetHandler(w http.ResponseWriter, r *http.Request) {
+	// Handle 'GET' method
+	// income ID-shorl url
+	// outcome Redirect by full url
+	if r.Method != http.MethodGet {
+		http.Error(w, "Only GET request are allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	id := chi.URLParam(r, "id")
+
+	if id == "" {
+		http.Error(w, "Bad request", http.StatusBadRequest)
+		return
+	}
+
+	// WORK with storage
+	// Move id - data url
+	// url, ok := storage[vars["id"]]
+	fullURL, ok := Store.Get(id)
+
+	if !ok {
+		http.Error(w, "Bad request", http.StatusBadRequest)
+		return
+	}
+	// log.Println("IDFefaultHandler func()\n\tOK is", ok, "\n\tItem is", fullURL)
+
+	w.Header().Add("Content-Type", "text/html; charset=UTF-8")
+	w.Header().Set("Location", string(fullURL))
+	w.WriteHeader(307)
+}
+
+func ShortenerURLHandler(w http.ResponseWriter, r *http.Request) {
+	// Handle POST method
+	// income full URL
+	// outcome None
+	// set up pair 'full url' - 'id' in storage
+	if r.Method != http.MethodPost {
+		http.Error(w, "Bad request", http.StatusBadRequest)
+		return
+	}
+	// Read URL
+	fullURL, err := io.ReadAll(r.Body)
+	// Handle error
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+	// WORK with full url
+	shortURL := shortener.RandStringBytes(7)
+	for _, ok := Store.Get(shortURL); ok; {
+		shortURL := shortener.RandStringBytes(7)
```

Comment:
```suggestion
		shortURL = shortener.RandStringBytes(7)
```

Если возникнет конфликт сгенерированных идентификаторов, то в текущей реализации все равно запишется первичный идентификатор, так как `:=` означает инициализацию новой переменной.